### PR TITLE
[sc-22642] GitHub Catalog Part 2

### PR DESCRIPTION
**Pull request by arooxa**
This catalog adds support for all entities and new relationships that result from the [GitHub Part 2 Story](https://app.shortcut.com/sgnl/story/22642/github-adapter-part-2-catalog-template)

### CODE REVIEW

File: github.sgnl.yaml [hunk 1]
```diff
@@ -2,15 +2,21 @@
...
 defaultSyncMinInterval: 1
 defaultApiCallFrequency: SECONDLY
 defaultApiCallMinInterval: 1
 type: "GitHub-1.0.0"
-adapterConfig: "ewogICAgImVudGVycHJpc2VTbHVnIjogInt7SW5wdXQgUmVxdWlyZWR9fSIKfQ==" # { "enterpriseSlug": "{{Input Required}}" }
+# Example Config:
+# {
+#    "enterpriseSlug": "{{Input Required}}",
+#    "isEntepriseCloud": "{{Input Required}}",
+#    "apiVersion": "{{Input Required for REST-based entities}}"
```
Comment (r-rakshith) [inline github.sgnl.yaml:17]: nit: We give this example comment so that the user has some idea about datatype.

```suggestion
#    "enterpriseSlug": "your-enterprise-slug",
#    "isEntepriseCloud": true / false,
#    "apiVersion": "v1"
```

File: github.sgnl.yaml [hunk 2]
```diff
@@ -640,6 +654,579 @@ entities:
...
+        type: Bool
+      - name: mergeStateStatus
+        externalId: mergeStateStatus
+        type: String
+      - name: mergeable
+        externalId: mergeable
+        type: String
+      - name: merged
+        externalId: merged
+        type: Bool
```
Comment (r-rakshith) [inline github.sgnl.yaml:752]: Should we index this? 
e.g. "Deny access to PROD if PR is merged".

File: github.sgnl.yaml [hunk 3]
```diff
@@ -640,6 +654,579 @@ entities:
...
+        type: DateTime
+      - name: number
+        externalId: number
+        type: Int64
+      - name: publishedAt
+        externalId: publishedAt
+        type: DateTime
+      - name: state
+        externalId: state
+        type: String
```
Comment (r-rakshith) [inline github.sgnl.yaml:764]: To index this?
"Allow access if PR state is open".

File: github.sgnl.yaml [hunk 4]
```diff
@@ -640,6 +654,579 @@ entities:
...
+      - name: committedDate
+        externalId: committedDate
+        type: DateTime
+      - name: commitAuthor
+        externalId: $.commit.author.user.id
+        type: String
+        indexed: true
+        attributeAlias: commitAuthorId    
+  # https://docs.github.com/en/graphql/reference/objects#pullrequestchangedfile
```
Comment (r-rakshith) [inline github.sgnl.yaml:909]:
```suggestion
  
  # https://docs.github.com/en/graphql/reference/objects#pullrequestchangedfile
```

File: github.sgnl.yaml [hunk 5]
```diff
@@ -640,6 +654,579 @@ entities:
...
+        attributeAlias: issueAuthorLogin
+      - name: repositoryId
+        externalId: $.repository.id
+        type: String
+        indexed: true
+        attributeAlias: issueRepositoryId
+      - name: activeLockReason
+        externalId: activeLockReason
+        type: String
+      - name: closed
```
Comment (r-rakshith) [inline github.sgnl.yaml:970]: To index?
"Deny access if assigned issue is closed"

File: github.sgnl.yaml [hunk 6]
```diff
@@ -640,6 +654,579 @@ entities:
...
+        externalId: issueId
+        type: String
+        indexed: true
+        attributeAlias: assignedIssueId
+      - name: userId
+        externalId: id
+        type: String
+        indexed: true
+        attributeAlias: issueAssigneeId
+      - name: login # remove later
```
Comment (r-rakshith) [inline github.sgnl.yaml:1037]: Is this required?
I see a "remove later" comment

File: github.sgnl.yaml [hunk 7]
```diff
@@ -640,6 +654,579 @@ entities:
...
+        externalId: issueId
+        type: String
+        indexed: true
+        attributeAlias: participantIssueId
+      - name: userId
+        externalId: id
+        type: String
+        indexed: true
+        attributeAlias: issueParticipantId
+      - name: login # remove later
```
Comment (r-rakshith) [inline github.sgnl.yaml:1062]: Is this required?
I see a "remove later" comment

File: github.sgnl.yaml [hunk 8]
```diff
@@ -640,6 +654,579 @@ entities:
...
+      - name: color
+        externalId: color
+        type: String
+      - name: createdAt
+        externalId: createdAt
+        type: DateTime
+      - name: isDefault
+        externalId: isDefault
+        type: Bool
+      - name: name
```
Comment (r-rakshith) [inline github.sgnl.yaml:1089]: To index?
"Allow access if `prod-bug` label is present"

File: github.sgnl.yaml [hunk 9]
```diff
@@ -640,6 +654,579 @@ entities:
...
+        externalId: name
+        type: String
+      - name: updatedAt
+        externalId: updatedAt
+        type: DateTime
+  RepositoryLabel: # TODO: Might want to remove this from catalog completely
```
Comment (r-rakshith) [inline github.sgnl.yaml:1095]:
```suggestion
  
  RepositoryLabel: # TODO: Might want to remove this from catalog completely
```

Comment (r-rakshith) [reply github.sgnl.yaml:1095]: Let's take a call on this before we merge this. Who will have answers to this?

File: github.sgnl.yaml [hunk 10]
```diff
@@ -640,6 +654,579 @@ entities:
...
+        externalId: repositoryId
+        type: String
+        indexed: true
+        attributeAlias: repositoryLabelRepositoryId
+      - name: labelId
+        externalId: id
+        type: String
+        indexed: true
+        attributeAlias: repositoryLabelId
+      - name: name # remove later
```
Comment (r-rakshith) [inline github.sgnl.yaml:1117]: Is this required?
I see a "remove later" comment

File: github.sgnl.yaml [hunk 11]
```diff
@@ -640,6 +654,579 @@ entities:
...
+        externalId: name
+        type: String
+  PullRequestLabel:
```
Comment (r-rakshith) [inline github.sgnl.yaml:1120]:
```suggestion
  
  PullRequestLabel:
```

File: github.sgnl.yaml [hunk 12]
```diff
@@ -640,6 +654,579 @@ entities:
...
+        externalId: pullRequestId
+        type: String
+        indexed: true
+        attributeAlias: pullRequestLabelPullRequestId
+      - name: labelId
+        externalId: labelId
+        type: String
+        indexed: true
+        attributeAlias: pullRequestLabelId
+      - name: name # remove later
```
Comment (r-rakshith) [inline github.sgnl.yaml:1142]: Is this required?
I see a "remove later" comment

File: github.sgnl.yaml [hunk 13]
```diff
@@ -640,6 +654,579 @@ entities:
...
+        externalId: name
+        type: String
+  IssueLabel:
```
Comment (r-rakshith) [inline github.sgnl.yaml:1145]:
```suggestion
  
  IssueLabel:
```

**Review by r-rakshith — CHANGES_REQUESTED** [review]
LGTM overall. Some formatting and styling comments
1. Please address the "remove later" comments throughout.
2. Add new line after every entity / relationship

@isabella-sgnl please have look at the default set of attributes that we want to index.

File: github.sgnl.yaml (hunk 10 above)
Comment (arooxa) [reply github.sgnl.yaml:1117]: Not required, I'm going to remove these 'name/login' attributes on the connection entities after testing in neo4j, it's just added in there for my convenience. I'll remove them before merging

File: github.sgnl.yaml (hunk 1 above)
Comment (arooxa) [reply github.sgnl.yaml:17]: fixed

File: github.sgnl.yaml (hunk 4 above)
Comment (arooxa) [reply github.sgnl.yaml:909]: fixed

File: github.sgnl.yaml (hunk 5 above)
Comment (arooxa) [reply github.sgnl.yaml:970]: added

File: github.sgnl.yaml (hunk 8 above)
Comment (arooxa) [reply github.sgnl.yaml:1089]: fixed

File: github.sgnl.yaml (hunk 2 above)
Comment (arooxa) [reply github.sgnl.yaml:752]: added

File: github.sgnl.yaml (hunk 3 above)
Comment (arooxa) [reply github.sgnl.yaml:764]: added

File: github.sgnl.yaml (hunk 10 above)
Comment (r-rakshith) [reply github.sgnl.yaml:1131]: Please maintain a copy of testing catalog in a different branch for personal/testing use. It is good practice to exclude testing code in pull requests (unless the test is a part of the PR).

File: github.sgnl.yaml [hunk 14]
```diff
@@ -640,6 +654,595 @@ entities:
...
+    attributes:
+      - name: number
+        externalId: number
+        type: Int64
+        indexed: true
+        uniqueId: true
+      - name: createdAt
+        externalId: created_at
+        type: DateTime
+      - name: state
```
Comment (isabella-sgnl) [inline github.sgnl.yaml:1203]: Let's index this attribute as we index state in Issues and other entities

File: github.sgnl.yaml [hunk 15]
```diff
@@ -726,24 +721,16 @@ entities:
       - name: isDraft
         externalId: isDraft
         type: Bool
-      - name: isInMergeQueue
-        externalId: isInMergeQueue
-        type: Bool
-      - name: isMergeQueueEnabled
-        externalId: isMergeQueueEnabled
-        type: Bool
```
Comment (r-rakshith) [inline github.sgnl.yaml]: Why are these removed?

**Review by r-rakshith — CHANGES_REQUESTED** [review]
LGTM overall. I see some attributes deleted. Any specific reason why?

File: github.sgnl.yaml (hunk 15 above)
Comment (arooxa) [reply github.sgnl.yaml]: These attributes were listed in the documentation, but they failed during testing. Looks like they've removed support for them.

Comment (r-rakshith) [reply github.sgnl.yaml]: Are they not present in both Cloud and On Prem versions?

**Review by r-rakshith — APPROVED** [review]

File: github.sgnl.yaml (hunk 15 above)
Comment (arooxa) [reply github.sgnl.yaml]: They are not present in Enterprise Server which is the instance I tested on. It could be present on Enterprise Cloud, but I won't know for sure without creating a new trial because my last one expired. BTW - I have seen attributes that are present in only one of the two instances in the past, but also attributes that are listed in the documentation but not supported by other deployment type. Do you want me to change/test anything here?

Comment (r-rakshith) [reply github.sgnl.yaml]: If this is tested for GH Enterprise Server, then let's rename the file accordingly before we merge. In case GH Cloud has more attributes, we'll create another catalog for it, with the extra attributes.

File: github.sgnl.yaml
Comment (r-rakshith) [inline github.sgnl.yaml:1]: Rename to `github-enterprise-server.sgnl.yaml`

**Review by r-rakshith — CHANGES_REQUESTED** [review]
Rename file

**Review by r-rakshith — APPROVED** [review]
